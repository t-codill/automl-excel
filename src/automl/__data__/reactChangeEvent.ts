// tslint:disable: no-any
export const reactChangeEvent: React.ChangeEvent<any> = {
    target: {}
} as unknown as React.ChangeEvent<any>;
