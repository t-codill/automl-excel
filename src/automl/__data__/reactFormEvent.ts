// tslint:disable: no-any
export const reactFormEvent: React.FormEvent<any> = {
    preventDefault(): void { return; }
} as unknown as React.FormEvent<any>;
