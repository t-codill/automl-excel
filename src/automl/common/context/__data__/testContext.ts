import { PageNames } from "../../PageNames";
import { Logger } from "../../utils/logger";
import { IWorkspaceContextProps, WorkspaceError } from "../IWorkspaceProps";
import { WorkspaceFlight } from "../WorkspaceFlight";

export function getLogCustomEventSpy(): jest.SpyInstance<ReturnType<Logger["logCustomEvent"]>> {
    const spy = jest.spyOn(Logger.prototype, "logCustomEvent");
    spy.mockImplementation(jest.fn());
    return spy;
}
let pageName: PageNames = PageNames.Unknown;
export const testContext: IWorkspaceContextProps = {
    get pageName(): PageNames { return pageName; },
    get theme(): string { return "light"; },
    discoverUrls: {},
    flight: new WorkspaceFlight(""),
    location: "eastus",
    resourceGroupName: "testResource",
    subscriptionId: "00000000-0000-0000-0000-000000000000",
    logger: new Logger("test"),
    getToken: () => "testToken",
    setPageName: (value: PageNames) => { pageName = value; },
    workspaceName: "testWorkSpace",
    setNavigationBarButtons(): void { return; },
    onError: (err: WorkspaceError) => { throw err; },
    clearErrors: () => { return; },
    setHeader: () => { return; },
    setLoading: () => { return; }
};
