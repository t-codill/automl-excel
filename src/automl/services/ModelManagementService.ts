import { AzureMachineLearningModelManagementService, AzureMachineLearningModelManagementServiceModels } from "@vienna/model-management";
import { IServiceBaseProps } from "./ServiceBase";
import { ServiceBaseNonArm } from "./ServiceBaseNonArm";

const version = "1.0";

export class ModelManagementService extends ServiceBaseNonArm<AzureMachineLearningModelManagementService> {
    constructor(props: IServiceBaseProps) {
        super(props, AzureMachineLearningModelManagementService, props.discoverUrls.modelmanagement);
    }

    public async createAsset(
        name: string,
        description: string,
        url: string
    ): Promise<AzureMachineLearningModelManagementServiceModels.Asset | undefined> {
        return this.send(async (client, abortSignal) => {
            return client.assets.create(
                version,
                this.props.subscriptionId,
                this.props.resourceGroupName,
                this.props.workspaceName,
                {
                    abortSignal,
                    asset: {
                        artifacts: [
                            {
                                prefix: new URL(url).pathname.replace(/^\/azureml\//, "")
                            }
                        ],
                        name,
                        description
                    }
                });
        });
    }

    public async registerModel(
        name: string,
        description: string,
        assetId: string,
        mimeType: string,
        runId: string | undefined,
        experimentName: string
    ): Promise<AzureMachineLearningModelManagementServiceModels.Model | undefined> {
        return this.send(async (client, abortSignal) => {
            return client.mLModels.register(
                {
                    name,
                    url: `aml://asset/${assetId}`,
                    mimeType,
                    description,
                    runId,
                    experimentName
                },
                version,
                this.props.subscriptionId,
                this.props.resourceGroupName,
                this.props.workspaceName,
                {
                    abortSignal
                });
        });
    }
}
