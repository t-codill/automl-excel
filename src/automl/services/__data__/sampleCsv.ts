export const sampleCsv = `a,b,c,d,e
1,2,3,4,5
1,2,3,4,5
1,2,3,4,5
1,2,3,4,5
1,2,3,4,5
`;

const largeCsvHeader = Array(50)
    .fill("aaaaa,bbbbb,ccccc,ddddd,eeeee")
    .join(",");

const largeCsvRow = Array(50)
    .fill("11111,22222,33333,44444,55555")
    .join(",");

export const largeCsv = `${largeCsvHeader}
${Array(50)
        .fill(largeCsvRow)
        .join("\n")}
`;
