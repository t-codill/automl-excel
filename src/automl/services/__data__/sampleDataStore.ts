import { AzureMachineLearningDatastoreManagementModels } from "@vienna/datastore";

export const sampleDataStore: Required<AzureMachineLearningDatastoreManagementModels.DataStoreDto> = {
    dataStoreType: "AzureBlob",
    name: "defaultDataStore",
    azureStorageSection: {
        accountKey: "key",
        accountName: "account",
        credentialType: "AccountKey",
        credential: "secret",
        containerName: "defaultContainer",
        isSas: false
    },
    azureDataLakeSection: {},
    azurePostgreSqlSection: {},
    azureSqlDatabaseSection: {},
    hasBeenValidated: true
};
