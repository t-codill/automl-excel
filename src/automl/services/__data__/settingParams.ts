import { AdvancedSetting } from "../AdvancedSetting";

export const settingParams: AdvancedSetting = {
    jobType: "classification",
    column: "1",
    metric: "AUC_weighted",
    experimentTimeoutMinutes: 60,
    maxIteration: 15,
    experimentExitScore: 0.99,
    maxConcurrent: 1,
    maxCores: 2,
    preprocess: true,
    nCrossValidations: 5,
    blacklistAlgos: null,
    validationSize: 0.2,
    grainColumns: null,
    maxHorizon: null,
    timeSeriesColumn: null
};
