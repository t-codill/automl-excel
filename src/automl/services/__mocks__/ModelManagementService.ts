import { AzureMachineLearningModelManagementServiceModels } from "@vienna/model-management";

export class ModelManagementService {

    public async createAsset(): Promise<AzureMachineLearningModelManagementServiceModels.Asset | undefined> {
        return {
            id: "asset-id",
            name: "asset-name"
        };
    }

    public async registerModel(): Promise<AzureMachineLearningModelManagementServiceModels.Model | undefined> {
        return {
            id: "model-id",
            name: "model-name",
            url: "http://www.dummyurl.com/dummy",
            mimeType: "model-mimeType"
        };
    }

    public dispose(): void {
        return;
    }

    public reset(): void {
        return;
    }
}
