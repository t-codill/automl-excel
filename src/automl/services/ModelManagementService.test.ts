import { testContext } from "../common/context/__data__/testContext";
import { ModelManagementService } from "./ModelManagementService";

jest.mock("@vienna/model-management");

const service = new ModelManagementService(testContext);

describe("ModelManagementService", () => {
    it("should registerModel", async () => {
        const result = await service.registerModel("test", "test description", "test url", "test mime type", "experimentName", "runId");
        expect(result)
            .toMatchSnapshot();
    });
    it("should createAsset", async () => {
        const result = await service.createAsset("test", "test description", "http://www.dummyurl.com/dummy");
        expect(result)
            .toMatchSnapshot();
    });
});
