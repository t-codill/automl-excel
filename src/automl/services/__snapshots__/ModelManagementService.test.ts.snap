// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModelManagementService should createAsset 1`] = `
Object {
  "_mockParameters": Array [
    "1.0",
    "00000000-0000-0000-0000-000000000000",
    "testResource",
    "testWorkSpace",
    Object {
      "abortSignal": Object {},
      "asset": Object {
        "artifacts": Array [
          Object {
            "prefix": "/dummy",
          },
        ],
        "description": "test description",
        "name": "test",
      },
    },
  ],
  "id": "id",
  "mimeType": "mimeType",
  "name": "name",
  "url": "http://www.dummyurl.com/dummy",
}
`;

exports[`ModelManagementService should registerModel 1`] = `
Object {
  "_mockParameters": Array [
    Object {
      "description": "test description",
      "experimentName": "runId",
      "mimeType": "test mime type",
      "name": "test",
      "runId": "experimentName",
      "url": "aml://asset/test url",
    },
    "1.0",
    "00000000-0000-0000-0000-000000000000",
    "testResource",
    "testWorkSpace",
    Object {
      "abortSignal": Object {},
    },
  ],
  "id": "id",
  "mimeType": "mimeType",
  "name": "name",
  "url": "http://www.dummyurl.com/dummy",
}
`;
